test/integration/src/main/java/io/pravega/test/integration/selftest/adapters/BookKeeperAdapter.java

```diff
@@ -168,24 +174,18 @@ protected void shutDown() {
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public CompletableFuture<Void> append(String logName, Event event, Duration timeout) {
         ensureRunning();
-        LedgerHandle lh = this.ledgers.getOrDefault(logName, null);
+        WriteHandle lh = this.ledgers.getOrDefault(logName, null);
         if (lh == null) {
             return Futures.failedFuture(new StreamSegmentNotExistsException(logName));
         }
 
         ArrayView s = event.getSerialization();
-        val result = new CompletableFuture<Void>();
-        AsyncCallback.AddCallback addCallback = (rc, handle, entryId, ctx) -> {
-            if (rc == BKException.Code.OK) {
-                result.complete(null);
-            } else {
-                result.completeExceptionally(BKException.create(rc));
-            }
-        };
-        lh.asyncAddEntry(s.array(), s.arrayOffset(), s.getLength(), addCallback, null);
-        return result;
+        // appendAsync returns CompletableFuture<Long> 
+        return (CompletableFuture) lh
```

Comment:
Use `Futures.toVoid`. I do not like how this casting works here; this will also eliminate the need for `@SuppressWarnings`.

Reply:
@andreipaduroiu done